test/AsyncAPI.Saunter.Generator.Cli.Tests/PackAndInstallLocalTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Shouldly;
+using Xunit.Abstractions;
+
+namespace AsyncAPI.Saunter.Generator.Cli.Tests;
+
+public class PackAndInstallLocalTests(ITestOutputHelper output)
+{
+    private string Run(string file, string args, string workingDirectory, int expectedExitCode = 0)
+    {
+        var process = Process.Start(new ProcessStartInfo(file)
+        {
+            Arguments = args,
+            WorkingDirectory = workingDirectory,
+            RedirectStandardOutput = true,
+            RedirectStandardError = true,
+            UseShellExecute = false,
+        });
+        process.WaitForExit(TimeSpan.FromSeconds(20));
+        var stdOut = process.StandardOutput.ReadToEnd().Trim();
+        var stdError = process.StandardError.ReadToEnd().Trim();
+        output.WriteLine($"### Output of \"{file} {args}\"");
+        output.WriteLine(stdOut);
+        output.WriteLine(stdError);
+
+        process.ExitCode.ShouldBe(expectedExitCode);
+        return stdOut;
+    }
+
+    [Fact]
+    public void Pack_Install_Run_Uninstall_Test()
```

Comment:
it looks like an attempt to test the dotnet utility, not our solution.

also this test makes changes to my device, which is also wrong

Reply:
You are right. Would you be ok with this test if I change it to install the tool --local instead of --global?

In the Generator.Build follow-up branch, I had issues with this test and made it more robust,
Ref: https://github.com/asyncapi/saunter/pull/219/files#diff-367a8bef3834e73695852cab366ff3582f658d329d34b8aef41b275ec5277e8dR43

Reply:
it's better not to do e2e at all, try to isolate the logic and cover it with units. if there is no logic, then do everything on the test environment.

I've never tested utilities like this, I can't tell you with a link - here's an example of how it's done for the backend (via WebApplicationFactory) - https://learn.microsoft.com/en-us/aspnet/core/test/integration-tests ?view=aspnetcore-8.0